pkg/api/job_runs.go

```diff
@@ -116,6 +118,78 @@ func FetchJobRun(dbc *db.DB, jobRunID int64, logger *log.Entry) (*models.ProwJob
 	return jobRun, jobRunTestCount, nil
 }
 
+func findReleaseMatchJobNames(dbc *db.DB, jobRun *models.ProwJobRun, compareRelease string, logger *log.Entry) ([]string, int, error) {
```

Comment:
I think I can see what this is doing but I am not immediately clear why, could you add godoc or PR description to outline the change being made and the algorithm to do so, and maybe provide some small real world examples. I'm unclear about the segment matching as well as how variants continue to play into it.